inst/examples/sir.cpp

```diff
@@ -23,17 +23,18 @@ class sir {
     return ret;
   }
 
-  void update(size_t step, const std::vector<double> state,
-              dust::RNG<double, int>& rng, std::vector<double>& state_next) {
+  void update(size_t step, const std::vector<real_t>& state,
+              dust::rng_state_t<real_t>& rng_state,
+              std::vector<real_t>& state_next) {
     double S = state[0];
     double I = state[1];
     double R = state[2];
     double N = S + I + R;
 
     double p_SI = 1 - std::exp(-(data_.beta) * I / N);
     double p_IR = 1 - std::exp(-(data_.gamma));
-    double n_IR = rng.rbinom(round(I), p_IR * data_.dt);
-    double n_SI = rng.rbinom(round(S), p_SI * data_.dt);
+    double n_IR = dust::distr::rbinom(rng_state, I, p_IR * data_.dt);
```

Comment:
Why did the `round` go?

Reply:
Was in the generated odin code (where it's generally needed) but it's not actually needed here so I dropped it in pursuit of a simpler example